core/codec/rle/rle_plus.hpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CODEC_RLE_PLUS_HPP
+#define CODEC_RLE_PLUS_HPP
+
+#include "common/outcome.hpp"
+#include "codec/rle/rle_plus_errors.hpp"
+#include "codec/rle/rle_plus_decoding_stream.hpp"
+#include "codec/rle/rle_plus_encoding_stream.hpp"
+
+namespace fc::codec::rle {
+  /**
+   * @brief RLE+ encode
+   * @tparam T - type of elements to encode
+   * @tparam A - std::set allocator
+   * @param input - data to encode
+   * @return Encoded byte-vector
+   */
+  template <typename T, typename A>
+  std::vector<uint8_t> RLEPlusEncode(const std::set<T, A> &input) {
+    RLEPlusEncodingStream encoder;
+    encoder << input;
+    return encoder.data();
+  }
+
+  /**
+   * @brief RLE+ decode
+   * @tparam T - type of elements to decode
+   * @param input - data to decode
+   * @return Decoded data
+   */
+  template <typename T>
+  outcome::result<std::set<T>> RLEPlusDecode(gsl::span<const uint8_t> input) {
```

Comment:
Should `input` be passed by const ref?

Reply:
My opinion: we don't need to use a `gsl::span` reference, because this type is pretty small (just pointer to start and sequence length) and copy operations are very cheap. 
I think, the closest notion of `gsl::span` is `std::string_view` - this type also recommended to be passed by value instead reference.